status/reporter_gitlab.go

```diff
@@ -151,18 +152,24 @@ func (r *GitLabReporter) setCommitStatus(report TestReport) error {
 		opt.TargetURL = gitlab.Ptr(url)
 	}
 
-	// Special case for gitLab `running` state because of a bug where it can't be updated to the same state again
-	if glState == gitlab.Running {
+	// Fetch commit statuses only if necessary
+	if glState == gitlab.Running || glState == gitlab.Pending {
 		allCommitStatuses, _, err := r.client.Commits.GetCommitStatuses(r.sourceProjectID, r.sha, nil)
 		if err != nil {
 			return fmt.Errorf("error while getting all commitStatuses for sha %s: %w", r.sha, err)
 		}
-		existingCommitStatus := r.GetExistingCommitStatus(allCommitStatuses, report.FullName)
-		if existingCommitStatus != nil && existingCommitStatus.Status == string(gitlab.Running) {
-			r.logger.Info("Will not update the existing commit status from `running` to `running`",
-				"scenario.name", report.ScenarioName, "commitStatus.ID", existingCommitStatus.ID)
-			return nil
-		}
+		existingCommitStatus = r.GetExistingCommitStatus(allCommitStatuses, report.FullName)
+	}
+
+	// special case, we want to skip updating commit status if the status from running to running, from enque to pending
+	if (glState == gitlab.Running && existingCommitStatus != nil && existingCommitStatus.Status == string(gitlab.Running)) ||
```

Comment:
nit: it seems like this long condition could be simplified if it was moved under [here](https://konflux-ci.dev/docs/how-tos/testing/integration/choosing-contexts/), wdyt?